core/src/main/scala/com/gu/tableversions/core/TableVersions.scala

```diff
@@ -0,0 +1,56 @@
+package com.gu.tableversions.core
+
+import java.time.Instant
+
+import com.gu.tableversions.core.TableVersions.CommitResult
+
+/**
+  * This defines the interface for querying and updating table version information tracked by the system.
+  */
+trait TableVersions[F[_]] {
```

Comment:
I'm thinking `rollback(partition: List[Partition])` (to set the provided partitions to the previous version) will be a pretty widely used feature. Do you think it should be a part of this core API?

Reply:
Absolutely! But I didn't write out the specs for that scenario in this PR as I just aimed to get the basic scenarios working first.